cmd/prometheus/main.go

```diff
@@ -611,11 +615,26 @@ func main() {
 		discoveryManagerNotify = legacymanager.NewManager(ctxNotify, log.With(logger, "component", "discovery manager notify"), legacymanager.Name("notify"))
 	}
 
+	opts := promql.EngineOpts{
+		Logger:                   log.With(logger, "component", "query engine"),
+		Reg:                      prometheus.DefaultRegisterer,
```

Comment:
Do you think this would work out? it feels like we would need so somehow wrap the metrics.